pkgs/utils/utils.go

```diff
@@ -103,3 +104,70 @@ func GetNonce(input []byte) []byte {
 	ret := sha256.Sum256(input)
 	return ret[:]
 }
+
+// JoinSets creates a set of two groups of operators. For example: [1,2,3,4] and [1,2,5,6,7] will return [1,2,3,4,5,6,7]
+func JoinSets(oldOperators, newOperators []*spec.Operator) []*spec.Operator {
+	tmp := make(map[uint64]*spec.Operator)
+	var set []*spec.Operator
+	for _, op := range oldOperators {
+		if tmp[op.ID] == nil {
+			tmp[op.ID] = op
+		}
+	}
+	for _, op := range newOperators {
+		if tmp[op.ID] == nil {
+			tmp[op.ID] = op
+		}
+	}
+	for _, op := range tmp {
+		set = append(set, op)
+	}
+	return set
+}
+
+// GetCommonOldOperators returns an old set of operators disjoint from new set
+// For example: old set [1,2,3,4,5]; new set [3,4,5,6,7]; returns [3,4,5]
+func GetCommonOldOperators(oldOperators, newOperators []*spec.Operator) []*spec.Operator {
+	tmp := make(map[uint64]*spec.Operator)
+	var set []*spec.Operator
+	for _, op := range newOperators {
+		if tmp[op.ID] == nil {
+			tmp[op.ID] = op
+		}
+	}
+	for _, op := range oldOperators {
+		if tmp[op.ID] != nil {
+			set = append(set, op)
+		}
+	}
+	return set
+}
+
+// GetDisjointNewOperators returns a new set of operators disjoint from old set
+// For example: old set [1,2,3,4,5]; new set [3,4,5,6,7]; returns [6,7]
+func GetDisjointNewOperators(oldOperators, newOperators []*spec.Operator) []*spec.Operator {
+	tmp := make(map[uint64]*spec.Operator)
+	var set []*spec.Operator
+	for _, op := range newOperators {
+		if tmp[op.ID] == nil {
+			tmp[op.ID] = op
+		}
+	}
+	for _, op := range oldOperators {
+		if tmp[op.ID] != nil {
+			delete(tmp, op.ID)
+		}
+	}
+	for _, op := range tmp {
+		set = append(set, op)
+	}
+	return set
+}
```

Comment:
we're missing unit tests for these 3 funcs

these are important to test because they're quite hard to verify just by looking (imo)

Reply:
Okay agree. Ill make unit tests for these.